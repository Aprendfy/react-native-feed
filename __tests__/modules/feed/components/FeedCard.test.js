import React from 'react';
import { shallow } from 'enzyme';
import {
  FeedCard,
  CardHeader,
  CardBody,
} from '../../../../src/modules/feed/components';

const props = {
  title: 'title',
  tag: 'tag',
  readingTime: 'readingtime',
  level: 'level',
  color: 'red',
  image: 'image',
  category: 'category',
  body: 'BODY'
};

describe('<FeedCard />', () => {
  const wrapper = shallow(<FeedCard {...props} />);

  it('Should have a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render correctly', () => {
    const { title, category, readingTime, level, color, image, body } = props;
    const headerProps = { title, readingTime, level, color, image, category };

    const cardHeader = wrapper.find(CardHeader);
    const cardBody = wrapper.find(CardBody);

    expect(cardHeader.exists()).toBe(true);
    expect(cardBody.exists()).toBe(true);

    expect(cardHeader.props()).toEqual(headerProps);
    expect(cardBody.props()).toEqual({ text: body });
  });

});
