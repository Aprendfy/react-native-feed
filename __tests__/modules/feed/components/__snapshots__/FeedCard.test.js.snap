// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedCard /> Should have a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "flex": 1,
          }
    }
>
    <CardHeader
        category="category"
        color="red"
        image="image"
        level="level"
        readingTime="readingtime"
        title="title"
    />
    <CardBody
        text="BODY"
    />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "flex": 1,
                }
      }
>
      <CardHeader
            category="category"
            color="red"
            image="image"
            level="level"
            readingTime="readingtime"
            title="title"
      />
      <CardBody
            text="BODY"
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <FeedCard
        body="BODY"
        category="category"
        color="red"
        image="image"
        level="level"
        readingTime="readingtime"
        tag="tag"
        title="title"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": FeedCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "body": "BODY",
          "category": "category",
          "color": "red",
          "image": "image",
          "level": "level",
          "readingTime": "readingtime",
          "tag": "tag",
          "title": "title",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
>
          <CardHeader
                    category="category"
                    color="red"
                    image="image"
                    level="level"
                    readingTime="readingtime"
                    title="title"
          />
          <CardBody
                    text="BODY"
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
>
          <CardHeader
                    category="category"
                    color="red"
                    image="image"
                    level="level"
                    readingTime="readingtime"
                    title="title"
          />
          <CardBody
                    text="BODY"
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FeedCard
    body="BODY"
    category="category"
    color="red"
    image="image"
    level="level"
    readingTime="readingtime"
    tag="tag"
    title="title"
/>,
}
`;
